main.py

```diff
@@ -1,4 +1,280 @@
+            if word_all.lower() != '':
+                all_words_frequency_dict[word_all.lower()] += 1
+        for word_cap in word_list_full:
+            if word_cap.capitalize() == word_cap:
+                if word_cap.lower() != '':
+                    capitalized_words_dict[word_cap.lower()] += 1
+        word_dict_combined = {key: [all_words_frequency_dict[key.lower()],
+                                    capitalized_words_dict[key.lower()]] for key in all_words_frequency_dict}
+        return word_dict_combined
+
+
+# Class provides functionality to load fb2 file information to the database.
+class DBWriter:
+
+    @logger.catch()
+    def create_connection(self, db_file):
+        conn = None
+        try:
+            conn = sqlite3.connect(db_file)
+            logger.info('Connection to the database is established.')
+            return conn
+        except Error as e:
+            logger.error(e)
+        return conn
+
+    @logger.catch()
+    def close_connection(self, connection):
+        try:
+            connection.close()
+            logger.info('Connection is closed.')
+        except Error as e:
+            logger.error(e)
+
+    @logger.catch()
+    def create_table(self, conn, create_table_sql):
+        try:
+            c = conn.cursor()
+            c.execute(create_table_sql)
+        except Error as e:
+            logger.error(e)
+
+    @logger.catch()
+    def check_book_availability(self, conn, find_book_sql):
+        try:
+            c = conn.cursor()
+            c.execute(find_book_sql)
+        except Error as e:
+            logger.error(e)
+
+    @logger.catch()
+    def create_words_frequency_table(self, conn, book_name, word_dict):
+        sql_create_projects_table = """ CREATE TABLE IF NOT EXISTS '%s' (
+                                                    word text,
+                                                    count integer,
+                                                    count_uppercase integer
+                                                ); """ % book_name
+        # create table if not exists
+        if conn is not None:
+            # create projects table
+            self.create_table(conn, sql_create_projects_table)
+            logger.info("Book table is created")
+        else:
+            logger.error("Cannot create the database connection.")
+
+        # insert data
+        try:
+            columns = ', '.join("`" + str(x).replace('/', '_') + "`" for x in word_dict.keys())
+            values = ', '.join("'" + str(x).replace('/', '_') + "'" for x in word_dict.values())
+            c = conn.cursor()
+            row = c.execute(" SELECT 1 from '%s'" % book_name).fetchone()
+            if not row:
+                for key, values in word_dict.items():
+                    c.execute("""INSERT INTO '%s' (word, count, count_uppercase) VALUES (?, ?, ?)""" % book_name,
+                              [key, values[0], values[1]])
+                conn.commit()
```

Comment:
Не до конца понятно, зачем нужно делать "SELECT 1".

```suggestion
                    for key, values in word_dict.items():
                        c.execute("""INSERT INTO '%s' (word, count, count_uppercase) VALUES (?, ?, ?)""" % book_name,[key, values[0], values[1]])
                    conn.commit()
```

Reply:
С точки зрения перфоманса, иногда предпочтительние использовать ф-ю executemany, которая может производить batch операции. Подробнее можно почитать здесь: https://www.kite.com/python/docs/sqlite3.Cursor.executemany

Reply:
Эта функция проверяет, добавлена ли книга в таблицу. Если результат будет не пустой (селект единицы, чтобы по перформансу не проседал + нам не надо больше одной строки для этой проверки), значит книга уже добавлена. Если пустой - то добавляем ее в таблицу